File: 4. Java/EX44.java
package Apostila;

import java.util.Locale;
import java.util.Scanner;

public class EX44
{
    public static void main(String[] args) 
    {
        // Inicializa variáveis
        double acuAlt = 0; // Acumulador de altura
        int qtd, idAux = 0, contIda = 0; // Contadores e índices
        
        // Configura a localidade
        Locale.setDefault(Locale.US);
        
        // Lê a quantidade de pessoas
        try (Scanner ent = new Scanner(System.in))
        {
            // Lê a quantidade de pessoas
            System.out.print("# Quantas pessoas serao digitadas: ");            
            qtd = ent.nextInt();
            
            // Declara vetores para idade, altura e nomes
            int [] vetIda = new int[qtd];
            double [] vetAlt = new double[qtd];
            String [] vetNom = new String[qtd];
            String [] aux = new String[qtd]; // Para armazenar nomes dos menores de 16
            
            // Verifica o limite de pessoas
            while ((qtd < 1) || (qtd > 5))
            {
                System.out.println("# Valor maximo de 5 pessoas");
                System.out.print("# Quantas pessoas serao digitadas: ");
                qtd = ent.nextInt(); // Repete a entrada se fora do limite
            }
            
            // Lê os dados das pessoas
            for (int i = 0; i < qtd; i++)
            {
                System.out.printf("\n# Dados da %da pessoa:\n", i + 1);
                
                System.out.print("# Nome: ");
                vetNom[i] = ent.next();
                ent.nextLine(); // Limpa o buffer
                
                System.out.print("# Idade: ");
                vetIda[i] = ent.nextInt();
                
                System.out.print("# Altura: ");
                vetAlt[i] = ent.nextDouble();
                acuAlt += vetAlt[i]; // Acumula a altura
            }
            
            // Conta menores de 16 anos
            for (int i = 0; i < qtd; i++)
            {
                if (vetIda[i] < 16)
                {
                    contIda++; // Conta os menores de 16
                    aux[idAux] = vetNom[i]; // Armazena o nome
                    idAux++;
                }
            }
            
            // Exibe altura média e percentual de menores de 16
            System.out.printf("\n# Altura media: %.2f", (acuAlt / qtd));
            System.out.printf("\n# Pessoas com menos de 16 anos: %.1f%%\n", ((double) contIda / qtd) * 100);
            
            // Exibe nomes dos menores de 16
            for (int i = 0; i < contIda; i++)
            {
                System.out.println("# " + aux[i]); // Lista os nomes
            }
        }
    }
}

File: 6. Python/EX44.py
# Inicializa variáveis para acumular altura e controle de idade
acuAlt = 0
idAux = 0
contIda = 0

# Solicita a quantidade de pessoas a serem digitadas
qtd = int(input("# Quantas pessoas serao digitadas: "))

# Valida a entrada para garantir que esteja entre 1 e 5
while qtd < 1 or qtd > 5:
    print("# Valor maximo de 5 pessoas")
    qtd = int(input("# Quantas pessoas serao digitadas: "))

# Cria vetores para armazenar dados de idade, altura e nomes
vetIda = [0 for i in range(qtd)]
vetAlt = [0.0 for i in range(qtd)]
aux = ["" for i in range(qtd)]
vetNom = ["" for i in range(qtd)]

# Loop para coletar dados de cada pessoa
for i in range(qtd):
    print(f"\n# Dados da {i + 1}a pessoa: ")

    # Lê o nome, idade e altura da pessoa
    vetNom[i] = input("# Nome: ")
    vetIda[i] = int(input("# Idade: "))
    vetAlt[i] = float(input("# Altura: "))

    # Acumula a altura total
    acuAlt += vetAlt[i]

# Loop para contar pessoas com menos de 16 anos
for i in range(qtd):
    if vetIda[i] < 16:

        # Armazena nomes das pessoas menores de 16 anos
        aux[idAux] = vetNom[i]
        idAux += 1
        contIda += 1

# Exibe a altura média
print(f"# Altura media: {(acuAlt / qtd):.2f}")

# Exibe a porcentagem de pessoas com menos de 16 anos
print(f"# Pessoas com menos de 16 anos: {((contIda / qtd) * 100):.1f} %")

# Exibe os nomes das pessoas com menos de 16 anos
for i in range(contIda):
    print(aux[i])
